crates/genesis/src/lib.rs

```diff
@@ -293,15 +293,33 @@ fn deserialize_private_key<'de, D>(deserializer: D) -> Result<Option<B256>, D::E
 where
     D: Deserializer<'de>,
 {
-    let opt_str: Option<String> = Option::deserialize(deserializer)?;
-
-    if let Some(ref s) = opt_str {
-        if s == "0x" {
-            return Ok(None);
+    if deserializer.is_human_readable() {
+        match Option::<String>::deserialize(deserializer)? {
+            Some(ref s) => {
+                if s == "0x" {
+                    return Ok(None);
+                }
+                B256::from_str(s).map(Some).map_err(D::Error::custom)
+            }
+            None => Ok(None),
         }
-        B256::from_str(s).map(Some).map_err(D::Error::custom)
     } else {
-        Ok(None)
+        Option::<B256>::deserialize(deserializer)
+    }
+}
+
+/// Custom deserialization function for `Option<u64>`.
+///
+/// This function allows it to be deserialized form a number or a "quantity" hex string.
+/// We need a custom function as this should only be used for non-human-readable formats.
+fn deserialize_u64_opt<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    if deserializer.is_human_readable() {
+        alloy_serde::quantity::opt::deserialize(deserializer)
+    } else {
+        Option::<u64>::deserialize(deserializer)
     }
 }
```

Comment:
this function is a bit out of place here, same for the existing `deserialize_private_key` can we move both to end of file?

Reply:
done 08a6d86173f7c3bac25886fc0ca9a0612e195e72